mission/functions/systems/sites/fn_sites_create_aa_site.sqf

```diff
@@ -88,8 +88,21 @@ params ["_pos"];
 	//Teardown condition
 	{
 		params ["_siteStore"];
-		//Teardown when all guns destroyed
-		(_siteStore getVariable "aaGuns" findIf {alive _x} == -1)
+
+                private _pos = getPos _siteStore;
+                private _objects = _siteStore getVariable ["aaGuns", []];
+
+                /*
+                Teardown when all guns are either
+                - destroyed
+                - no longer with 20m radius of site centre point
+                */
+
+                private _objectsAreAliveInRadius = _objects findIf {
+                        (alive _x) && (count ([_x] inAreaArray [_pos, 20, 20, 0, false]) > 0)
+                };
+
+                _objectsAreAliveInRadius == -1
```

Comment:
It would be way faster to execute `inAreaArray` once, it also accepts an object as a centre pos, which I understand the `_siteStore` is (not familiar with MF internals).


Below code should be faster on average:
```suggestion
                /*
                Teardown when all guns are either
                - destroyed
                - no longer with 20m radius of site centre point
                */           
                ((_siteStore getVariable "aaGuns") inAreaArray [_siteStore, 20, 20, 0, false]) findIf {alive _x} == -1
```

Reply:
@veteran29 

- Extracted logic out to new script as per my comment https://github.com/Savage-Game-Design/Mike-Force/pull/180#issuecomment-2395169406
- Switched the ordering around for the 1x `inAreaArray` execution

---

I was working on an additional check for when players have loaded the site object into a vehicle with the Advanced Logistics module, as that can causes site/zone soft-locks too -- see: https://github.com/Savage-Game-Design/Mike-Force/pull/180/commits/cd97f17cb4c8143c9654abdd469d3f5f56f9101d

But there's at least three downstream edge cases to take into account, beyond the teardown condition check

- do not delete object when player is carrying object (delete object in front of player)
- do not delete object when in a vehicle logistics inventory (object no longer exists in logistics inventory)
- do not delete object if ever has been in a vehicle logistics inventory (delete object in front of player)

Simplest / cleanest option seems to be to just to disable the ability to load these objects into vehicles, as per latest.